logbat/src/main/java/info/logbat/domain/log/presentation/validation/LogRequestsValidator.java

```diff
@@ -0,0 +1,125 @@
+package info.logbat.domain.log.presentation.validation;
+
+import info.logbat.domain.log.presentation.payload.request.CreateLogRequest;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+/**
+ * 로그 요청 목록의 유효성을 검사하는 밸리데이터 클래스입니다. 빈 요청, 개별 요청의 유효성, 전체 요청의 유효성을 검사합니다.
+ */
+@Component
+@RequiredArgsConstructor
+public class LogRequestsValidator implements
+    ConstraintValidator<ValidLogRequests, List<CreateLogRequest>> {
+
+    private final Validator validator;
+
+    /**
+     * 로그 요청 목록의 유효성을 검사합니다.
+     *
+     * @param requests 검사할 로그 요청 목록
+     * @param context  제약 조건 컨텍스트
+     * @return 유효성 검사 결과 (true: 유효, false: 무효)
+     */
+    @Override
+    public boolean isValid(List<CreateLogRequest> requests, ConstraintValidatorContext context) {
+        if (isEmptyRequest(requests, context)) {
+            return false;
+        }
+
+        List<String> errorMessages = new ArrayList<>();
+        List<CreateLogRequest> validRequests = filterValidRequests(requests, errorMessages);
+
+        updateRequestsList(requests, validRequests);
+
+        if (requests.isEmpty()) {
+            addErrorMessage(context, String.join("\n", errorMessages));
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * 요청 목록이 비어있는지 확인합니다.
+     *
+     * @param requests 검사할 요청 목록
+     * @param context  제약 조건 컨텍스트
+     * @return 비어있으면 true, 그렇지 않으면 false
+     */
+    private boolean isEmptyRequest(List<CreateLogRequest> requests,
+        ConstraintValidatorContext context) {
+        if (requests == null || requests.isEmpty()) {
+            addErrorMessage(context, "빈 요청이 전달되었습니다.");
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * 유효한 요청만 필터링합니다.
+     *
+     * @param requests      전체 요청 목록
+     * @param errorMessages 오류 메시지를 저장할 리스트
+     * @return 유효한 요청 목록
+     */
+    private List<CreateLogRequest> filterValidRequests(List<CreateLogRequest> requests,
+        List<String> errorMessages) {
+        return requests.stream()
+            .filter(request -> isValidRequest(request, requests.indexOf(request), errorMessages))
+            .toList();
```

Comment:
`indexOf` 의 시간복잡도가 O(N)이라서 이렇게 직접 반복문을 돌 수 있도록 수정하는게 어떨까요?
아니면, index의 역할이 에러 message에만 추가하는 목적으로 보여서 index를 사용하지 않고 message를 추가하는 편도 좋겠어요!
```suggestion
        List<CreateLogRequest> validRequests = new ArrayList<>();
        for (int i = 0; i < requests.size(); i++) {
            CreateLogRequest request = requests.get(i);
            if (isValidRequest(request, i, errorMessages)) {
                validRequests.add(request);
            }
        }
        return validRequests;
```

Reply:
좋습니다~~

Reply:
@LuizyHub @tidavid1 확인 한 번 부탁드려요~

[caf8bea](https://github.com/woowa-techcamp-2024/Team5-Guys/pull/115/commits/caf8beac6e4f1345b269abb1b43b60697b2ac9aa)